commerce-product-service/src/main/java/com/liferay/commerce/product/internal/upgrade/v1_11_0/CPAttachmentFileEntryGroupUpgradeProcess.java

```diff
@@ -0,0 +1,125 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.commerce.product.internal.upgrade.v1_11_0;
+
+import com.liferay.asset.kernel.model.AssetCategory;
+import com.liferay.asset.kernel.service.AssetCategoryLocalService;
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.exception.NoSuchClassNameException;
+import com.liferay.portal.kernel.exception.NoSuchGroupException;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.Statement;
+
+/**
+ * @author Alec Sloan
+ */
+public class CPAttachmentFileEntryGroupUpgradeProcess extends UpgradeProcess {
+
+	public CPAttachmentFileEntryGroupUpgradeProcess(
+		AssetCategoryLocalService assetCategoryLocalService,
+		ClassNameLocalService classNameLocalService) {
+
+		_assetCategoryLocalService = assetCategoryLocalService;
+		_classNameLocalService = classNameLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (Statement s = connection.createStatement();
+			ResultSet rs = s.executeQuery(
+				"select classNameId, classPK from CPAttachmentFileEntry")) {
+
+			long cpDefinitionClassNameId = _getCPDefinitionClassNameId();
+			long assetCategoryClassNameId =
+				_classNameLocalService.getClassNameId(AssetCategory.class);
+
+			String updateCPAttachmentFileEntrySQL =
+				"update CPAttachmentFileEntry set groupId = ? where " +
+					"classNameId = ? and classPK = ?";
+
+			try (PreparedStatement ps =
+					AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+						connection, updateCPAttachmentFileEntrySQL)) {
+
+				while (rs.next()) {
+					long classNameId = rs.getLong("classNameId");
+					long classPK = rs.getLong("classPK");
+
+					long groupId;
+
+					if (classNameId == assetCategoryClassNameId) {
+						AssetCategory assetCategory =
+							_assetCategoryLocalService.getAssetCategory(
+								classPK);
+
+						groupId = assetCategory.getGroupId();
+					}
+					else if (classNameId == cpDefinitionClassNameId) {
+						groupId = _getGroupIdFromCPDefinition(classPK);
+					}
+					else {
+						throw new NoSuchClassNameException();
+					}
+
+					ps.setLong(0, groupId);
+					ps.setLong(1, classNameId);
+					ps.setLong(2, classPK);
+
+					ps.executeUpdate();
+				}
+
+				ps.executeBatch();
+			}
+		}
+	}
+
+	private long _getCPDefinitionClassNameId() throws Exception {
+		try (Statement s = connection.createStatement();
+			ResultSet rs = s.executeQuery(
+				"select classNameId from ClassName_ where value = " +
+					"'com.liferay.commerce.product.model.CPDefinition'")) {
+
+			if (rs.next()) {
+				return rs.getLong("classNameId");
+			}
+		}
+
+		throw new NoSuchClassNameException();
+	}
```

Comment:
why can't you use `_classNameLocalService` for this?

Reply:
because it would have to be on CPDefinition.class.... but that should be fine since it's the same module right?

Reply:
yeah we can use the CPDefinition class, just not any services registered by this module.